lib/get-request-agent.js

```diff
@@ -17,7 +16,13 @@ function getRequestAgent (options) {
   agentOptions.forEach(option => {
     console.warn(`options.${option} is deprecated. Use "options.agent" instead`)
   })
-  const Agent = require(options.protocol || DEFAULTS.protocol).Agent
+
+  var Agent
+  if (options.protocol && options.protocol === 'http') {
+    Agent = require('http').Agent
+  } else {
+    Agent = require('https').Agent
+  }
```

Comment:
I think style-wise I would prefer to just load both on top

```js
const HttpAgent = require('http').Agent
const HttpsAgent = require('https').Agent
```

or 

```js
const AGENTS = {
  http: require('http').Agent
  https: require('https').Agent
}
```

Do you see any reason not to do it that way?